File: features/step_definitions/tag_steps.rb
# frozen_string_literal: true

Given(/^I have the following tags$/) do |tags|
  tags.hashes.each do |tag|
    send "create_#{tag['LOCATION']}_tag", tag['NAME']
  end
end


Given(/^I have a remote tag "([^"]+)" that is not on a branch$/) do |name|
  create_standalone_remote_tag name
end




Then(/^I now have the following tags$/) do |expected_tags|
  expected_tags.diff! TagFinder.all_tags.to_table
end

File: test/steps/tag_steps.go
package steps

import (
	"fmt"

	"github.com/cucumber/godog"
	"github.com/cucumber/messages-go/v10"
)

// TagSteps defines Gherkin step implementations around merges.
func TagSteps(suite *godog.Suite, state *ScenarioState) {
	suite.Step(`^my repo has the following tags$`, func(table *messages.PickleStepArgument_PickleTable) error {
		return state.gitEnv.CreateTags(table)
	})

	suite.Step(`^my repo now has the following tags$`, func(table *messages.PickleStepArgument_PickleTable) error {
		tagTable, err := state.gitEnv.TagTable()
		if err != nil {
			return err
		}
		diff, errorCount := tagTable.EqualGherkin(table)
		if errorCount != 0 {
			fmt.Printf("\nERROR! Found %d differences in the existing tags\n\n", errorCount)
			fmt.Println(diff)
			return fmt.Errorf("mismatching tags found, see diff above")
		}
		return nil
	})

	suite.Step(`^my repo has a remote tag "([^"]+)" that is not on a branch$`, func(name string) error {
		return state.gitEnv.OriginRepo.CreateStandaloneTag(name)
	})
}
